SDK/MobileCenter/Microsoft.Azure.Mobile.Windows.Shared/Storage/Storage.cs

```diff
@@ -261,31 +233,40 @@ internal Storage(IStorageAdapter storageAdapter)
         /// <exception cref="StorageException"/>
         public async Task<string> GetLogsAsync(string channelName, int limit, List<Log> logs)
         {
-            await OpenDbAsync().ConfigureAwait(false);
+            await StartTaskAsync().ConfigureAwait(false);
             logs?.Clear();
             var retrievedLogs = new List<Log>();
             MobileCenterLog.Debug(MobileCenterLog.LogTag, $"Trying to get up to {limit} logs from storage for {channelName}");
             try
```

Comment:
>try [](start = 12, length = 3)

General feedback: I'm not a big fan of a big try-catch block that covers more than 40 lines of code plus multiple operations, except in test code. It makes the code less maintainable and less readable. I'd like to see this try-catch block refactored into shorter ones.